packages/wdio-allure-reporter/src/reporter.ts

```diff
@@ -505,88 +573,135 @@ export default class AllureReporter extends WDIOReporter {
+            this.onTestStart(hook)
         }
-
-        // add hook as test to suite
-        this.onTestStart(hook)
     }
 
     onHookEnd(hook: HookStats) {
-        const { disableMochaHooks, useCucumberStepReporter } = this._options
+        const { disableHooks, useCucumberStepReporter } = this._options
 
-        // ignore global hooks
+        // ignore global hooks or hooks when option is set in false
+        // any hook is skipped if there is not a suite created.
         if (!hook.parent || !this._state.currentSuite) {
             return
         }
 
-        const isMochaAllHook = isMochaAllHooks(hook.title)
-        const isMochaEachHook = isMochaEachHooks(hook.title)
-
-        if (disableMochaHooks && !isMochaAllHook && !this._state.currentAllureTestOrStep) {
-            return
-        }
-
-        // set beforeEach / afterEach hook (step) status
-        if (disableMochaHooks && isMochaEachHook) {
-            this._endTest(hook.error ? AllureStatus.FAILED : AllureStatus.PASSED, hook.error)
-            // mocha doesn't fire onTestEnd for failed beforeEach hook
-            // we have to force mark test as failed test
-            if (isMochaBeforeEachHook(hook.title) && hook.error) {
-                const fakeTestStat = Object.assign({}, hook, { title: hook.currentTest }) as TestStats
-                this._endTest(fakeTestStat.error ? AllureStatus.FAILED : AllureStatus.PASSED, fakeTestStat.error)
+        const isAllHook = isAllTypeHooks(hook.title) // if the hook is before* for mocha/jasmine
+        const isEachHook = isEachTypeHooks(hook.title) // if the hook is after* for mocha/jasmine
+
+        // if the hook is before/after from mocha/jasmine
+        if ((isAllHook || isEachHook) && !disableHooks) {
+            // getting the root step
+            const currentHookRootStep = this._state.pop()
+            // getting before/after hook element
+            const currentHookRoot = this._state.pop()
+            if (currentHookRootStep || currentHookRoot) {
+                if (currentHookRootStep instanceof AllureStep && currentHookRoot instanceof ExecutableItemWrapper) {
+                    // stage to finish for all hook.
+                    currentHookRoot.stage = currentHookRootStep.stage = Stage.FINISHED
+                    // set status values
+                    switch (hook.state) {
+                    case 'passed':
+                        currentHookRoot.status = currentHookRootStep.status = Status.PASSED
+                        break
+                    case 'failed':
+                        currentHookRoot.status = currentHookRootStep.status = Status.FAILED
+                        break
+                    default:
+                        currentHookRoot.status = currentHookRootStep.status = Status.BROKEN
+                    }
+                    // set error data
+                    currentHookRoot.detailsMessage = currentHookRootStep.detailsMessage = hook.error?.message
```

Comment:
The error object must pass through `getErrorFromFailedTest(err)' to properly format errors from different test runners. e.g. :

```
const error = getErrorFromFailedTest(isFailed)
entity.detailsMessage = error.message
entity.detailsTrace = error.stack
```